File: C++/Remove-Nth-Node-From-End-of-List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *dummy = new ListNode(0);
        dummy->next = head;
        ListNode *prev = dummy;
        int count = 0;
        while (count < n) {
            prev = prev->next;
            count ++;
        }
        ListNode *cur = dummy;
        while (prev->next != NULL) {
            prev = prev->next;
            cur = cur->next;
        }
        cur->next = cur->next->next;
        return dummy->next;
    }
};

File: Python/Remove-Nth-Node-From-End-of-List.py
#!/usr/bin/env python


#Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


# @return a ListNode
def removeNthFromEnd(self, head, n):
    node1 = head
    for i in range(n):
        node1 = node1.next
    if node1 == None:
        return head.next
    node2 = head
    while node1.next != None:
        node1 = node1.next
        node2 = node2.next
    node2.next = (node2.next).next
    return head